providers/user.rb

```diff
@@ -0,0 +1,87 @@
+use_inline_resources if defined?(use_inline_resources)
+
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+action :upsert do
```

Comment:
@amsdard yeah, your right. logic itself is upsert. however, from the action perspective, user might be confused the name.